lib/apiservers/engine/backends/cache/layer_cache.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"sync"
+)
+
+// LCache is an in-memory cache to account for existing image layers
+// It is used primarily by imagec when coordinating layer downloads
+// The cache is initially hydrated by way of the image cache at startup
```

Comment:
why do we have a cache? Can't we use fs semantics to see the status of the download? We are already streaming the file into the fs (which is tmpfs) so we could do what downloaders are doing (create the lock file, download the file, move etc.) No?